### DISCUSSION TITLE
Unknow error with `sns.violinplot` when selecting only one ‘x’ attribute

### DISCUSSION THREAD

**Discussion by songyuc** [mwaskom/seaborn#3028] [Q&A]
Hi, guys,
I am learning using `seaborn` today, and met an unknow error with `sns.violinplot` when selecting only one ‘x’ attribute as,
```python
import seaborn as sns
sns.set_theme(style="whitegrid")

# Load the example tips dataset
tips = sns.load_dataset("tips")

# Draw a nested violinplot and split the violins for easier comparison
sns.violinplot(data=tips[tips["day"]=="Fri"], x="day", y="total_bill", hue="smoker",
               inner="quart", linewidth=1,split=True,
               palette={"Yes": "b", "No": ".85"})
sns.despine(left=True)
```  
And here was the error message,
<blockquote>
C:\Users\46630\anaconda3\envs\conda-general\lib\site-packages\seaborn\categorical.py:1261: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  if j and any(self.plot_hues[0] == hue_level):
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [5], in <cell line: 8>()
      5 tips = sns.load_dataset("tips")
      7 # Draw a nested violinplot and split the violins for easier comparison
----> 8 sns.violinplot(data=tips[tips["day"]=="Fri"], x="day", y="total_bill", hue="smoker",
      9                inner="quart", linewidth=1,split=True,
     10                palette={"Yes": "b", "No": ".85"})
     11 sns.despine(left=True)

File ~\anaconda3\envs\conda-general\lib\site-packages\seaborn\categorical.py:2312, in violinplot(data, x, y, hue, order, hue_order, bw, cut, scale, scale_hue, gridsize, width, inner, split, dodge, orient, linewidth, color, palette, saturation, ax, **kwargs)
   2309 if ax is None:
   2310     ax = plt.gca()
-> 2312 plotter.plot(ax)
   2313 return ax

File ~\anaconda3\envs\conda-general\lib\site-packages\seaborn\categorical.py:1419, in _ViolinPlotter.plot(self, ax)
   1417 def plot(self, ax):
   1418     """Make the violin plot."""
-> 1419     self.draw_violins(ax)
   1420     self.annotate_axes(ax)
   1421     if self.orient == "h":

File ~\anaconda3\envs\conda-general\lib\site-packages\seaborn\categorical.py:1261, in _ViolinPlotter.draw_violins(self, ax)
   1255     self.draw_stick_lines(ax, violin_data,
   1256                           support, density, i,
   1257                           ["left", "right"][j])
   1259 # The box and point interior plots are drawn for
   1260 # all data at the group level, so we just do that once
-> 1261 if j and any(self.plot_hues[0] == hue_level):
   1262     continue
   1264 # Get the whole vector for this group level

TypeError: 'bool' object is not iterable
</blockquote>
Could please tell me how to fix it?

Your answer and guide will be appreciated!

**Comment by mwaskom** [comment]
I think this is a subtle bug in seaborn that specifically happens with (A) split violins and (B) categorical grouping variable. If you disable the split you can see that it is still trying to draw the other levels (this is by design, there's just a bug where it doesn't handle empty data in those levels properly):

![image](https://user-images.githubusercontent.com/315810/190624589-b99c40aa-3507-468a-9d99-986fc3cc8ce0.png)

What I would do is not worry about selecting rows from the dataframe and instead just ask for a specific level using `order`:

```python
sns.violinplot(
    data=tips,
    x="day", y="total_bill", hue="smoker",
    order=["Fri"],  # <--- Add this
    inner="quart", linewidth=1, split=True,
    palette={"Yes": "b", "No": ".85"}
)
```

![image](https://user-images.githubusercontent.com/315810/190624817-d3ea5cf4-e452-439c-9736-2043a88fc3ed.png)
